homeassistant/components/renault/sensor.py

```diff
@@ -0,0 +1,277 @@
+
+
+class RenaultBatteryTemperatureSensor(RenaultBatteryDataEntity, SensorEntity):
+    """Battery Temperature sensor."""
+
+    _attr_device_class = DEVICE_CLASS_TEMPERATURE
+    _attr_unit_of_measurement = TEMP_CELSIUS
+
+    @property
+    def state(self) -> int | None:
+        """Return the state of this entity."""
+        return self.data.batteryTemperature if self.data else None
+
+
+class RenaultChargeModeSensor(RenaultChargeModeDataEntity, SensorEntity):
+    """Charge Mode sensor."""
+
+    _attr_device_class = DEVICE_CLASS_CHARGE_MODE
+
+    @property
+    def state(self) -> str | None:
+        """Return the state of this entity."""
+        return self.data.chargeMode if self.data else None
+
+    @property
+    def icon(self) -> str:
+        """Icon handling."""
+        if self.data and self.data.chargeMode == "schedule_mode":
+            return "mdi:calendar-clock"
+        return "mdi:calendar-remove"
+
+
+class RenaultChargeStateSensor(RenaultBatteryDataEntity, SensorEntity):
+    """Charge State sensor."""
+
+    _attr_device_class = DEVICE_CLASS_CHARGE_STATE
+
+    @property
+    def state(self) -> str | None:
+        """Return the state of this entity."""
+        charging_status = self.data.get_charging_status() if self.data else None
+        return slugify(charging_status.name) if charging_status is not None else None
+
+    @property
+    def icon(self) -> str:
+        """Icon handling."""
+        return "mdi:flash" if self.is_charging else "mdi:flash-off"
+
+
+class RenaultChargingRemainingTimeSensor(RenaultBatteryDataEntity, SensorEntity):
+    """Charging Remaining Time sensor."""
+
+    _attr_icon = "mdi:timer"
+    _attr_unit_of_measurement = TIME_MINUTES
+
+    @property
+    def state(self) -> int | None:
+        """Return the state of this entity."""
+        return self.data.chargingRemainingTime if self.data else None
+
+
+class RenaultChargingPowerSensor(RenaultBatteryDataEntity, SensorEntity):
+    """Charging Power sensor."""
+
+    _attr_device_class = DEVICE_CLASS_ENERGY
+    _attr_unit_of_measurement = POWER_KILO_WATT
+
+    @property
+    def state(self) -> float | None:
+        """Return the state of this entity."""
+        if not self.data or self.data.chargingInstantaneousPower is None:
+            return None
+        if self.vehicle.details.reports_charging_power_in_watts():
+            # Need to convert to kilowatts
+            return self.data.chargingInstantaneousPower / 1000
+        return self.data.chargingInstantaneousPower
+
+
+class RenaultFuelAutonomySensor(RenaultCockpitDataEntity, SensorEntity):
+    """Fuel autonomy sensor."""
+
+    _attr_icon = "mdi:gas-station"
+    _attr_unit_of_measurement = LENGTH_KILOMETERS
+
+    @property
+    def state(self) -> int | None:
+        """Return the state of this entity."""
+        return (
+            round(self.data.fuelAutonomy)
+            if self.data and self.data.fuelAutonomy is not None
+            else None
```

Comment:
We prefer regular if/else blocks if the ternary operator expression spans more than one line.